src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerReference.cs

```diff
@@ -62,6 +62,12 @@ public virtual string Display
         /// <param name="language">Language name.</param>
         public abstract ImmutableArray<DiagnosticAnalyzer> GetAnalyzers(string language);
 
+        /// <summary>
+        /// Tries to get the cached list of analyzers for this assembly reference for the given <paramref name="language"/>, if
+        /// it has already been created and is still cached.
+        /// </summary>
+        public abstract bool TryGetAnalyzers(string language, out ImmutableArray<DiagnosticAnalyzer> analyzers);
```

Comment:
Is it a "better" API change to put this in as `virtual`, and just have it return false? I don't know if we expect 3rd party inheritors of this.

Reply:
Since this is a public type, we can't make this abstract. The safest approach would be this:

```suggestion
        public virtual bool TryGetAnalyzers(string language, out ImmutableArray<DiagnosticAnalyzer> analyzers)
        {
            analyzers = GetAnalyzers(language);
            return true;
        }
```

Reply:
Having this call `GetAnalyzers` and return true means it doesn't match the semantics of the the `TryXXX` for things like `TryGetCompilation`, and means the version on `AnalyzerFileReference` wouldn't match the semantics of the base class.

Reply:
I don't think adding a `TryGet` API on `AnalyzerReference` is a good idea. That indicates lazyiness/statefulness, which is not true and cannot be enforced on sub-types. Any new `TryGet` API, if required, should directly on `AnalyzerFileReference`.